src/components/missions/joinButton.js

```diff
@@ -0,0 +1,38 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { Button } from 'react-bootstrap';
+import { joinMission, leaveMission } from '../../redux/missions/missions';
+import style from './missions.css';
+
+const JoinButton = ({ status, id }) => {
+  const dispatch = useDispatch();
+  const fields = {
+    'Not a member': {
+      variant: 'outline-secondary',
+      message: 'Join Mission',
+      click: () => { dispatch(joinMission(id)); },
+    },
+    'Active member': {
+      variant: 'outline-danger',
+      message: 'Leave Mission',
+      click: () => { dispatch(leaveMission(id)); },
+    },
+  };
```

Comment:
- [ ] Rockets/Dragons and Missions should use the React conditional rendering syntax, please do so to comply with the project requirements:

```javascript
{mission.reserved && ( 
    // render Leave Mission button
)}
```